modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/CollectionListRowAdapter.kt

```diff
@@ -62,31 +113,68 @@ internal class CollectionListRowAdapter(
         val inflater = LayoutInflater.from(parent.context)
         val binding = ItemCollectionListBinding.inflate(inflater, parent, false)
 
-        return CollectionListViewHolder(binding) { pageIndex, podcastIndex ->
-            val podcastSublist = getItem(pageIndex)
-            val podcast = podcastSublist.getOrNull(podcastIndex) as? DiscoverPodcast
+        return CollectionListViewHolder(
+            binding,
+            onItemClicked = { pageIndex, podcastIndex ->
+                val podcastSublist = getItem(pageIndex)
+                val podcast = podcastSublist.getOrNull(podcastIndex) as? DiscoverPodcast
+
+                if (podcast == null) return@CollectionListViewHolder
+
+                fromListId?.let {
+                    analyticsTracker.track(
+                        AnalyticsEvent.DISCOVER_LIST_PODCAST_TAPPED,
+                        mapOf(
+                            LIST_ID_KEY to it,
+                            PODCAST_UUID_KEY to podcast.uuid,
+                        ),
+                    )
+                }
+                onPodcastClicked(podcast, fromListId)
+            },
+            onHeaderClicked = {
+                onHeaderClicked()
+            },
+            onPodcastSubscribe = { pageIndex, podcastIndex ->
+                val podcastSublist = getItem(pageIndex)
+                val podcast = podcastSublist.getOrNull(podcastIndex) as? DiscoverPodcast
+                podcast?.let { onPodcastSubscribe(it, fromListId) }
+            },
+        )
+    }
 
-            if (podcast == null) return@CollectionListViewHolder
+    override fun onBindViewHolder(holder: CollectionListViewHolder, position: Int) {
+        val podcastSublist = getItem(position)
 
-            fromListId?.let {
-                analyticsTracker.track(
-                    AnalyticsEvent.DISCOVER_LIST_PODCAST_TAPPED,
-                    mapOf(
-                        LIST_ID_KEY to it,
-                        PODCAST_UUID_KEY to podcast.uuid,
-                    ),
-                )
+        holder.bind(podcastSublist, isSubscribeButtonVisible = position != 0)
+
+        if (position == 0) {
+            this.header?.let {
+                holder.binding.header.root.isVisible = true
+                holder.binding.header.lblTitle.text = it.title
+                holder.binding.header.lblSubtitle.text = it.subtitle
+                holder.binding.header.imageHeader.load(it.imageUrl)
+                holder.binding.header.root.contentDescription =
+                    it.title + " " + holder.binding.root.context.getString(LR.string.discover_collection_header_content_description_suffix)
```

Comment:
Should we be doing the concatenation in the strings file rather than here? Like the following:
```
<string name="podcast_next_episode_value">Next episode %s</string>
```